File: syscisepro/FORMULARIOS/CONTABILIDAD/PORCENTAJES RETENCION/FrmBuscarPlanCuentas.vb
Imports ClassLibraryCisepro.ENUMS
Imports ClassLibraryCisepro.CONTABILIDAD.PLAN_DE_CUENTAS
Imports syscisepro.DATOS

Namespace FORMULARIOS.CONTABILIDAD.PORCENTAJES_RETENCION
    ''' <summary>
    ''' CISEPRO - SEPORTPAC - ASENAVA 2019
    ''' Para buscar cuentas en PLAN DE CUENTAS
    ''' </summary>
    Public Class FrmBuscarPlanCuentas
        Private _tipoCon As TipoConexion
        Property TipoCox As Integer
            Get
                Select Case _tipoCon
                    Case TipoConexion.Asenava
                        Return 2
                    Case TipoConexion.Seportpac
                        Return 1
                    Case Else
                        Return 0
                End Select
            End Get
            Set(ByVal value As Integer)
                Select Case value
                    Case 2
                        _tipoCon = TipoConexion.Asenava
                    Case 1
                        _tipoCon = TipoConexion.Seportpac
                    Case Else
                        _tipoCon = TipoConexion.Cisepro
                End Select
            End Set
        End Property
        Public Tipo As Integer
        Private _objpLClassPlanDeCuentas As New ClassPlanDeCuentas

        Private Sub FrmBuscarPlanCuentas_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
            ' DEFINIR TIPO Y COLOR DE SISTEMA
            Select Case _tipoCon
                Case TipoConexion.Asenava
                    Icon = My.Resources.logo_a
                Case TipoConexion.Seportpac
                    Icon = My.Resources.logo_s
                Case Else
                    Icon = My.Resources.logo_c
            End Select

            BackColor = ValidationForms.GetColorSistema(_tipoCon)
            Label1.ForeColor = Color.White
            label2.ForeColor = Color.White
            cbxTipoCuenta.SelectedIndex = 0
            txtParametrobusqueda.Focus()
        End Sub

        Private Sub txtParametrobusqueda_KeyUp(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles txtParametrobusqueda.KeyUp
            If txtParametrobusqueda.Text.Trim().Length = 0 Then Return
            CargarCuentas(txtParametrobusqueda.Text.Trim())
        End Sub

        Private Sub CargarCuentas(ByVal fil As String)
            Try

                Dim data As DataTable
                listView1.Items.Clear()
                listView1.Groups.Clear()

                Select Case Tipo
                    Case 1
                        listView1.Groups.Add(New ListViewGroup("ACTIVOS"))
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(_tipoCon, False, 1, fil)
                    Case 2
                        listView1.Groups.Add(New ListViewGroup("PASIVOS"))
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(_tipoCon, False, 2, fil)
                    Case 3
                        listView1.Groups.Add(New ListViewGroup("PATRIMONIO"))
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(_tipoCon, False, 3, fil)
                    Case 4
                        listView1.Groups.Add(New ListViewGroup("INGRESOS"))
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(_tipoCon, False, 4, fil)
                    Case 5
                        listView1.Groups.Add(New ListViewGroup("GASTOS"))
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(_tipoCon, False, 5, fil)
                    Case 6
                        listView1.Groups.Add(New ListViewGroup("GANANCIAS"))
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(_tipoCon, False, 6, fil)
                    Case Else
                        listView1.Groups.Add(New ListViewGroup("ACTIVOS"))
                        listView1.Groups.Add(New ListViewGroup("PASIVOS"))
                        listView1.Groups.Add(New ListViewGroup("PATRIMONIO"))
                        listView1.Groups.Add(New ListViewGroup("INGRESOS"))
                        listView1.Groups.Add(New ListViewGroup("GASTOS"))
                        listView1.Groups.Add(New ListViewGroup("GANANCIAS"))

                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(_tipoCon, True, 0, fil)
                End Select


                For Each row As DataRow In data.Rows
                    Dim lst As New ListViewItem(If(row.Item(0) IsNot Nothing, row.Item(0).ToString, String.Empty), GetListViewGroup(row.Item(7).ToString.Trim))
                    lst.UseItemStyleForSubItems = False
                    For i = 1 To data.Columns.Count - 1
                        lst.SubItems.Add(If(row.Item(i) IsNot Nothing, row.Item(i).ToString, String.Empty))
                    Next
                    listView1.Items.Add(lst)
                Next

                For Each col As ColumnHeader In listView1.Columns
                    If col.Width > 5 Then col.Width = -2
                Next

                label4.Visible = False
            Catch ex As Exception
                label4.Text = "ERROR AL CARGAR DATOS: " & ex.Message
                label4.Visible = True
                listView1.Clear()
            End Try
        End Sub

        Private Sub listView1_DoubleClick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles listView1.DoubleClick
            If listView1.Items.Count = 0 Then Return
            If listView1.SelectedItems.Count = 0 Then Return
            DialogResult = DialogResult.OK
        End Sub

        Private Function GetListViewGroup(ByVal name As String) As ListViewGroup
            Dim grup As ListViewGroup = Nothing
            For Each grupo As ListViewGroup In From grupo1 As ListViewGroup In ListView1.Groups Where grupo1.ToString.Trim.Equals(name.Trim)
                grup = grupo
                Exit For
            Next
            Return grup
        End Function

        Private Sub cbxTipoCuenta_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cbxTipoCuenta.SelectedIndexChanged
            Tipo = cbxTipoCuenta.SelectedIndex
            CargarCuentas(txtParametrobusqueda.Text.Trim())
        End Sub
    End Class
End Namespace

File: SysCisepro3/Contabilidad/FrmBuscarPlanCuentas.cs
using System;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Windows.Forms;
using ClassLibraryCisepro3.Contabilidad.PlanDeCuentas;
using ClassLibraryCisepro3.Enums;
using SysCisepro3.Datos;
using SysCisepro3.Properties;
using Krypton.Toolkit;

namespace SysCisepro3.Contabilidad
{
    public partial class FrmBuscarPlanCuentas : KryptonForm
    {
        /// <summary>
        /// CISEPRO 2019
        /// Para buscar cuentas en PLAN DE CUENTAS
        /// </summary>
        private readonly ClassPlanDeCuentas _objpLClassPlanDeCuentas;
        public TipoConexion TipoCon { get; set; }
        public int Tipo { get; set; }

        public FrmBuscarPlanCuentas()
        {
            InitializeComponent();
            _objpLClassPlanDeCuentas = new ClassPlanDeCuentas();
        }

        private void FrmBuscarPlanCuentas_Load(object sender, EventArgs e)
        {
            // CARGAR ICONO Y DESIEÑO SEGUN SISTEMA 

            BackColor = ValidationForms.GetColorSistema(TipoCon);
            Label1.ForeColor = Color.White;
            label2.ForeColor = Color.White;
            switch (TipoCon)
            {
                case TipoConexion.Seportpac:
                    Icon = Resources.logo_s;
                    break;
                case TipoConexion.Asenava:
                   Icon = Resources.logo_a; 
                    break;
                default: // CISEPRO
                    Icon = Resources.logo_c; 
                    break;
            }

            cbxTipoCuenta.SelectedIndex = 0;
            txtParametrobusqueda.Focus();
        }

        private void txtParametrobusqueda_KeyUp(object sender, KeyEventArgs e)
        { 
            if (txtParametrobusqueda.Text.Trim().Length == 0) return;
            CargarCuentas(txtParametrobusqueda.Text.Trim());
        }

        private void CargarCuentas(string fil)
        {
            try
            {
                DataTable data;
                listView1.Items.Clear();
                listView1.Groups.Clear();
                
                switch (Tipo)
                {
                    case 1: listView1.Groups.Add(new ListViewGroup("ACTIVOS"));
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(TipoCon, false, 1, fil);
                        break;
                    case 2: listView1.Groups.Add(new ListViewGroup("PASIVOS"));
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(TipoCon, false, 2, fil);
                        break;
                    case 3: listView1.Groups.Add(new ListViewGroup("PATRIMONIO"));
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(TipoCon, false, 3, fil);
                        break;
                    case 4: listView1.Groups.Add(new ListViewGroup("INGRESOS"));
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(TipoCon, false, 4, fil);
                        break;
                    case 5: listView1.Groups.Add(new ListViewGroup("GASTOS"));
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(TipoCon, false, 5, fil);
                        break;
                    case 6: listView1.Groups.Add(new ListViewGroup("GANANCIAS"));
                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(TipoCon, false, 6, fil);
                        break;
                    default:
                        listView1.Groups.Add(new ListViewGroup("ACTIVOS"));
                        listView1.Groups.Add(new ListViewGroup("PASIVOS"));
                        listView1.Groups.Add(new ListViewGroup("PATRIMONIO"));
                        listView1.Groups.Add(new ListViewGroup("INGRESOS"));
                        listView1.Groups.Add(new ListViewGroup("GASTOS"));
                        listView1.Groups.Add(new ListViewGroup("GANANCIAS"));

                        data = _objpLClassPlanDeCuentas.SeleccionarPlanCuentas(TipoCon, true, 0, fil);
                        break;
                }

                foreach (DataRow s in data.Rows)
                {
                    var li = new ListViewItem(s[0].ToString(), GetListViewGroup(s[7].ToString().Trim()));
                    for (var i = 1; i < data.Columns.Count - 1; i++) li.SubItems.Add(s[i].ToString());
                    listView1.Items.Add(li);
                }

                foreach (var col in listView1.Columns.Cast<ColumnHeader>().Where(col => col.Width > 5)) col.Width = -2;

                label4.Visible = false;
            }
            catch (Exception ex)
            {
                label4.Text = @"ERROR AL CARGAR DATOS: " + ex.Message;
                label4.Visible = true;
                listView1.Clear();
            }
        }

        private void ListView1_DoubleClick(object sender, EventArgs e)
        {
            if (listView1.Items.Count == 0) return;
            if (listView1.SelectedItems.Count == 0) return;

            DialogResult = DialogResult.OK;
        }

        private ListViewGroup GetListViewGroup(string gname)
        {
            var g = new ListViewGroup();
            foreach (var @group in listView1.Groups.Cast<ListViewGroup>().Where(@group => @group.Header.Equals(gname)))
            {
                g = @group;
            }
            return g;
        }

        private void cbxTipoCuenta_SelectedIndexChanged(object sender, EventArgs e)
        {
            Tipo = cbxTipoCuenta.SelectedIndex;
            CargarCuentas(txtParametrobusqueda.Text.Trim());
        }
    }
}